programs/uxd/src/instructions/mint_with_router.rs

```diff
@@ -0,0 +1,437 @@
+use crate::IDENTITY_DEPOSITORY_NAMESPACE;
+use crate::MERCURIAL_VAULT_DEPOSITORY_LP_TOKEN_VAULT_NAMESPACE;
+use crate::MERCURIAL_VAULT_DEPOSITORY_NAMESPACE;
+
+#[derive(Accounts)]
+pub struct MintWithRouter<'info> {
+    /// #1
+    pub user: Signer<'info>,
+
+    /// #2
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    /// #3
+    #[account(
+        mut,
+        seeds = [CONTROLLER_NAMESPACE],
+        bump = controller.load()?.bump,
+        constraint = controller.load()?.registered_mercurial_vault_depositories[0] == mercurial_vault_depository_0.key() @UxdError::InvalidDepository,
+        constraint = controller.load()?.registered_credix_lp_depositories[0] == credix_lp_depository_0.key() @UxdError::InvalidDepository,
+        has_one = redeemable_mint @UxdError::InvalidRedeemableMint
+    )]
+    pub controller: AccountLoader<'info, Controller>,
+
+    /// #4
+    #[account(mut)]
+    pub redeemable_mint: Box<Account<'info, Mint>>,
+
+    /// #5
+    pub collateral_mint: Box<Account<'info, Mint>>,
+
+    /// #6
+    #[account(
+        mut,
+        constraint = user_redeemable.owner == user.key() @UxdError::InvalidOwner,
+        constraint = user_redeemable.mint == redeemable_mint.key() @UxdError::InvalidRedeemableMint,
+    )]
+    pub user_redeemable: Box<Account<'info, TokenAccount>>,
+
+    /// #7
+    #[account(
+        mut,
+        constraint = user_collateral.owner == user.key() @UxdError::InvalidOwner,
+        constraint = user_collateral.mint == collateral_mint.key() @UxdError::InvalidCollateralMint,
+    )]
+    pub user_collateral: Box<Account<'info, TokenAccount>>,
+
+    /// #8 - UXDProgram on chain account bound to a Controller instance that represent the blank minting/redeeming
+    #[account(
+        mut,
+        seeds = [IDENTITY_DEPOSITORY_NAMESPACE],
+        bump = identity_depository.load()?.bump,
+    )]
+    pub identity_depository: AccountLoader<'info, IdentityDepository>,
+
+    /// #9 - Token account holding the collateral from minting
+    #[account(
+        mut,
+        seeds = [IDENTITY_DEPOSITORY_COLLATERAL_NAMESPACE],
+        token::authority = identity_depository,
+        token::mint = identity_depository.load()?.collateral_mint,
+        bump = identity_depository.load()?.collateral_vault_bump,
+    )]
+    pub identity_depository_collateral_vault: Box<Account<'info, TokenAccount>>,
+
+    /// #10
+    #[account(
+        mut,
+        seeds = [MERCURIAL_VAULT_DEPOSITORY_NAMESPACE, mercurial_vault_depository_0.load()?.mercurial_vault.key().as_ref(), mercurial_vault_depository_0.load()?.collateral_mint.as_ref()],
+        bump = mercurial_vault_depository_0.load()?.bump,
+        has_one = controller @UxdError::InvalidController,
+        has_one = collateral_mint @UxdError::InvalidCollateralMint,
+        constraint = mercurial_vault_depository_0.load()?.mercurial_vault == mercurial_vault_depository_0_vault.key() @UxdError::InvalidMercurialVault,
```

Comment:
Probably same here as previous comment, seems pretty convoluted

Reply:
Discussed offline, merging for now, with intention to fix later